enhancements/builds/node-cache-for-builds.md

```diff
@@ -0,0 +1,138 @@
+---
+title: node-cache-for-builds
+authors:
+  - "@adambkaplan"
+reviewers:
+  - "@nalind
+  - "@rhdan"
+  - "@TomSweeneyRedHat"
+  - "@bparees"
+approvers:
+  - "@bparees"
+creation-date: 2019-02-18
+last-updated: 2019-02-18
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Node Cache for Builds
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+1. Can cri-o be enhanced so pods are not scheduled if the kubelet/node garbage collection is in progress?
+
+## Summary
+
+Builds in OpenShift 4.x currently require every layer of the base image to be pulled from its
+source registry. In OCP 3.x builds could take advantage of image layers that were stored on the
+build's node. This proposal provides a means of restoring this node-level cache.
```

Comment:
This limitation also means we get no layer reuse when executing dockerfile commands.  Being able to use layers from disk could help us with that, though i don't think the solution we've proposed so far would actually get us that (the layers might exist on the node in the case that a previous build's output image had been pulled to the node for some reason, but buildah isn't going to see/use them during Dockerfile statement execution, i suspect.  Worth talking with @nalind about though, because being able to no-op a command like "RUN go build ." would be significant).

Reply:
Added a sentence here denoting this limitation.